src/app/Utilities.ts

```diff
@@ -66,10 +70,34 @@ const RandomString = () => {
     return result;
 }
 
+/* Function for returning all key value pairs from an object, including nested ones */
+const ReturnPropertiesFromNestedObject = (propertiesObject: Dict) => {
+    const propertiesArray: { key: string, value: string | number | boolean }[] = [];
+
+    const PushToPropertiesArray = (propertyPair: { key: string, value: string | number | boolean }) => {
+        propertiesArray.push(propertyPair);
+    }
+
+    const LoopOverProperties = (propertiesObject: Dict) => {
```

Comment:
Recursion looks fine :+1: What we could do is write a unit test for these type of functions just to make sure they work and keep working

Reply:
Will keep this in mind, might be good as its a pure functional function